Add component yaml configuration definition with connection env renaming guide (#1684)
wso2/docs-choreo-dev: en/docs/develop-components/manage-component-source-configurations.md

        To use custom environment variable names instead of Choreo's default ones, add the dependency as a `serviceReference` in your `component.yaml v1.1` file. You can copy the `serviceReference` section from the `component.yaml v1.0` tab and paste it under `dependencies` in your `component.yaml v1.1` file, which maintains backward compatibility with the v1.0 format.

    ### Runtime configurations

>     In the `configuration` section of the `component.yaml` file, you can define runtime configurations for the component. Currently, the configurations support env variable injection related to dependencies.

Reviewer: ```suggestion
    In the `configuration` section of the `component.yaml` file, you can define runtime configurations for the component. These configurations currently support environment variable injection related to dependencies.
```

    
    | Configuration                 | Required     | Description                                                                      |
    |-------------------------------|--------------|----------------------------------------------------------------------------------|

>     | **env**                       | Optional     | Array of env variable confiuration                                               |

Reviewer: ```suggestion
    | **env**                       | Optional     | An array of env variable configurations.                                               |
```

>     | **&nbsp;name**                | Required     | A unique name starting with a letter or an underscore, containing letters, numbers and underscore for the environment variable name |

Reviewer: ```suggestion
    | **&nbsp;name**                | Required     | A unique name for the environment variable, starting with a letter or an underscore, and containing only letters, numbers, or underscores. |
```

>     | **&nbsp;valueFrom**           | Required     | Value source for the env variable                                                |

Reviewer: Can we say "The source of the environment variable value."?

    | **&nbsp;&nbsp;connectionRef** | Required     | Connection reference value source definition                                     |

>     | **&nbsp;&nbsp;&nbsp;name**    | Required     | Choreo connection name to refer the value from                                   |

Reviewer: Can we say "The name of the Choreo connection to reference the value from."?

Author: Yes. I'll update.

>     | **&nbsp;&nbsp;&nbsp;key**     | Required     | Choreo connection configuration key to refer the value from. Refer to [connection configurations](https://wso2.com/choreo/docs/develop-components/sharing-and-reusing/use-a-connection-in-your-service/) or see in-line in-line developer guide displayed during connection creation to discover available keys                   |

Reviewer: Suggestion:
"The Choreo connection configuration key to reference the value from. For details on available keys, see [Consume a service through a connection](https://wso2.com/choreo/docs/develop-components/sharing-and-reusing/use-a-connection-in-your-service/#consume-a-service-through-a-connection) or the inline developer guide."

    !!! note

>         Runtime configurations are supported only from component.yaml v1.1.

Reviewer: ```suggestion
        Runtime configurations are supported starting from `component.yaml v1.1`.
```

>         When environment variable value source is specified using `connectionRef`, the connection environment variable will be renamed into the given environment name. For example, in above sample `component.yaml` file, `CHOREO_HR_CONNECTION_SERVICEURL` variable from the `hr-connection` will be renamed to `HR_SERVICE_URL`.

Reviewer: Suggestion:
When an environment variable value is specified using `connectionRef`, the connection's environment variable is renamed to the environment variable name defined in the `configuration` section. For example, in the sample `component.yaml` file given above, the `CHOREO_HR_CONNECTION_SERVICEURL` variable in the `hr-connection` is renamed to `HR_SERVICE_URL`.
